ui/app/src/modules/System/Sites/Edit/EditSiteDialog.tsx

```diff
@@ -0,0 +1,225 @@
+        <form>
+          <Grid container spacing={3}>
+            <Grid item xs={12}>
+              <TextField
+                id="name"
+                name="name"
+                label={
+                  <FormattedMessage
+                    id="editSiteDialog.siteName" defaultMessage="Site Name"
+                  />
+                }
+                fullWidth
+                onChange={(event) => onSiteNameChange(event.target.value)}
+                value={siteName}
+                inputProps={{ maxLength: 50 }}
+                autoFocus
+              />
+            </Grid>
+
+            <Grid item xs={12}>
+              <TextField
+                id="description"
+                name="description"
+                label={
+                  <FormattedMessage
+                    id="editSiteDialog.siteDescription" defaultMessage="Site Description"
+                  />
+                }
+                fullWidth
+                onChange={(event) => onSiteDescriptionChange(event.target.value)}
+                value={siteDescription??''}
+                inputProps={{ maxLength: 4000 }}
+              />
+            </Grid>
+          </Grid>
+        </form>
+      </DialogBody>
+      <DialogFooter>
+        {
+          onClose && (
+          <Button onClick={onClose} variant="contained">
+            <FormattedMessage id="editSiteDialog.cancel" defaultMessage="Cancel" />
+          </Button>
+        )}
+        {
+          onSubmit && (
+          <Button onClick={() => onSubmit()} variant="contained" color="primary" autoFocus disabled={submitting}>
+            <FormattedMessage id="editSiteDialog.continue" defaultMessage="Continue" />
+          </Button>
+        )}
+      </DialogFooter>
+    </>
+  );
+}
+
+function EditSiteDialog(props: EditSiteDialogProps) {
+  const {
+    site,
+    open,
+    onClosed,
+    onClose,
+    onSaveSuccess
+  } = props;
+  const [siteData, setSiteData] = useSpreadState({
+    id: '',
+    name: '',
+    description: ''
+  });
+  const [apiState, setApiState] = useSpreadState<ApiState>({
+    error: null,
+    submitting: false
+  })
+
+  useUnmount(onClosed);
+
+  const resource = useLogicResource<Return, Source>(site, {
+    shouldResolve: (source) => Boolean(source),
+    shouldReject: (source) => false,
+    shouldRenew: (source, resource) => resource.complete,
+    resultSelector: (source) => source,
+    errorSelector: (source) => null
+  });
+
+  useEffect(() => {
+    if (site) {
+      setSiteData(site);
+    }
+  }, [site, setSiteData]);
+
+  const onSiteNameChange = (name: string) => {
+    setSiteData({ name });
+  }
+
+  const onSiteDescriptionChange = (description: string) => {
+    setSiteData({ description });
+  }
+
+  const handleSubmit = () => {
+    setApiState({ submitting: true });
+
+    editSite(siteData).subscribe(
+      (response) => {
+        setApiState({ submitting: false });
+        onSaveSuccess?.(response);
+        onClose?.();
```

Comment:
don't call onClose here, it'd be the caller's choice to close it and it'd be doable through the onSaveSuccess